asmcli/lib/util.sh

```diff
@@ -42,6 +42,28 @@ run_command() {
   return $RETVAL
 }
 
+######
+# check_curl calls curl and returns non-zero if the http code is not 200 or the
+# curl command fails.
+######
+check_curl() {
+  local TMPFILE; TMPFILE=$(mktemp)
+  local HTTPCODE; HTTPCODE=$(run_command curl --write-out '%{http_code}' --silent --show-error --output "$TMPFILE" "${@}")
+  local RETVAL; RETVAL="$?"
+  if [[ "$RETVAL" != "0" ]] ; then
+    rm "$TMPFILE"
+    return "$RETVAL"
+  fi
+  if [[ "$HTTPCODE" != "200" ]] ; then
+    echo "HTTP response code: ${HTTPCODE}"
+  fi
+  cat "$TMPFILE" && rm "$TMPFILE"
+  if [[ "$HTTPCODE" != "200" ]] ; then
+    return 1
```

Comment:
Returning a specific error code implies that there's some meaning other than just "it didn't work". The canonical(?) way is to do `false; return`, which doesn't imply anything other than "not good".

Reply:
Done